File: Toy-Language-Interpreter_C#/Toy-Language-Interpreter/Model/Statement/AssignmentStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Toy_Language_Interpreter.DataTypes;
using Toy_Language_Interpreter.Model.Expression;
using Toy_Language_Interpreter.MyException;

namespace Toy_Language_Interpreter.Model.Statement
{
    class AssignmentStatement : IStatement
    {
        private string id;
        private AExpression expr;

        public AssignmentStatement(string id, AExpression expr)
        {
            this.id = id;
            this.expr = expr;
        }

        public ProgramState Execute(ProgramState state)
        {
            MyIDictionary<string, int> symTab = state.GetSymTab();
            try
            {
                int value = expr.Evaluate(symTab);
                symTab.Put(id, value);
            }
            catch (ExpressionEvaluationException e)
            {
                Console.WriteLine(e.ToString());
            }
            return null;
        }

        public override string ToString()
        {
            return id + " = " + expr.ToString();
        }
    }
}

File: Toy-Language-Interpreter/src/Model/Statement/AssignmentStatement.java
package Model.Statement;

import DataTypes.MyIDictionary;
import DataTypes.MyIHeap;
import Model.Expression.Expression;
import Model.ProgramState;
import Exception.*;

/**
 * Created by vladc on 12.10.2016.
 */
public class AssignmentStatement implements IStatement {
    private String      id;
    private Expression  expr;

    public AssignmentStatement(String id, Expression expr) {
        this.id = id;
        this.expr = expr;
    }

    @Override
    public String toString() {
        return id + " = " + expr.toString();
    }

    public ProgramState execute(ProgramState state) throws MyStatementExecutionException {
        MyIDictionary<String, Integer> symTab = state.getSymTable();
        MyIHeap heap = state.getHeap();

        try {
            int value = expr.evaluate(symTab, heap);
            symTab.put(id, value);
        }
        catch (ExpressionEvaluationException ex) {
            ex.printStackTrace();
        }
        return null;
    }
}
